### PR TITLE
Make custom nvJEPG allocator return a relevant allocation status

### PR DESCRIPTION
**Pull request by JanuszL**
- fixes lack of returning a proper allocation status in nvJPEG custom allocator. So far it always returns 0.

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It  fixes lack of returning a proper allocation status in nvJPEG custom allocator. So far it always returns 0.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     make custom nvJEPG allocator return a relevant allocation status
 - Affected modules and functionalities:
     nvJPEG based decoder
 - Key points relevant for the review:
     NA
 - Validation and testing:
     CI
 - Documentation (including examples):
     NA

**JIRA TASK**: *[DALI-1721]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

File: dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_memory.cc [hunk 1]
```diff
@@ -205,12 +205,12 @@ void DeleteAllBuffers(std::thread::id thread_id) {
 
 static int DeviceNew(void **ptr, size_t size) {
   *ptr = GetBuffer(std::this_thread::get_id(), AllocType::GPU, size);
-  return 0;
+  return *ptr != nullptr ? cudaSuccess : cudaErrorMemoryAllocation;
```
Comment (mzient) [inline dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_memory.cc:208]: What happens if requested size is 0?

Comment (JanuszL) [reply dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_memory.cc:212]: Adjusted in the lines above.

**Comment by JanuszL** [comment]
!build

**Review by mzient — APPROVED** [review]

**Pull request merged**